File: libandroid-navigation/src/test/java/org/maplibre/navigation/android/navigation/v5/location/replay/ReplayRouteLocationConverterTest.java
package org.maplibre.navigation.android.navigation.v5.location.replay;

import org.maplibre.geojson.LineString;
import org.maplibre.geojson.Point;

import org.junit.Test;
import org.maplibre.navigation.android.navigation.v5.location.replay.ReplayRouteLocationConverter;

import java.util.List;

public class ReplayRouteLocationConverterTest {


    @Test
    public void testSliceRouteWithEmptyLineString() {
        ReplayRouteLocationConverter replayRouteLocationConverter = new ReplayRouteLocationConverter(null, 100, 1);
        List<Point> result = replayRouteLocationConverter.sliceRoute(LineString.fromJson(""));

        assert (result.isEmpty());
    }

    @Test
    public void testSliceRouteWithNullLineString() {
        ReplayRouteLocationConverter replayRouteLocationConverter = new ReplayRouteLocationConverter(null, 100, 1);
        List<Point> result = replayRouteLocationConverter.sliceRoute(null);

        assert (result.isEmpty());
    }
}

File: libandroid-navigation/src/test/java/org/maplibre/navigation/android/navigation/v5/location/replay/ReplayRouteLocationConverterTest.kt
package org.maplibre.navigation.android.navigation.v5.location.replay

import org.junit.Test
import org.maplibre.geojson.LineString

class ReplayRouteLocationConverterTest {
    @Test
    fun testSliceRouteWithEmptyLineString() {
        val replayRouteLocationConverter = ReplayRouteLocationConverter(null, 100, 1)
        val result = replayRouteLocationConverter.sliceRoute(LineString.fromJson(""))

        assert(result.isEmpty())
    }

    @Test
    fun testSliceRouteWithNullLineString() {
        val replayRouteLocationConverter = ReplayRouteLocationConverter(null, 100, 1)
        val result = replayRouteLocationConverter.sliceRoute(null)

        assert(result.isEmpty())
    }
}